core/ledger/pvtdatastorage/store.go

```diff
@@ -816,7 +825,10 @@ func (s *Store) processCollElgEvents() {
 	s.purgerLock.Lock()
 	defer s.purgerLock.Unlock()
 	collElgStartKey, collElgEndKey := createRangeScanKeysForCollElg()
-	eventItr := s.db.GetIterator(collElgStartKey, collElgEndKey)
+	eventItr, err := s.db.GetIterator(collElgStartKey, collElgEndKey)
+	if err != nil {
+		panic(err)
```

Comment:
Here and below, I would replace `panic` by an error return and then log the error in the caller. `panic` would be too harsh for a situation that neither causes all the functionality to be unavailable nor causes some data corruption. If we want to take more noticeable step, we could cause the channel to be declared as unusable but that may require more changes than what is in the scope of this PR.
So, I would just look at this PR something that improves the current code by logging the error (otherwise, the current behavior is not changed at all).

Reply:
I do not have a strong opinion on this. I agree that only functionality that gets affected is that the new collection member would not get the private data. This could leads to endorsement failures due to missing pvtdata in the stateDB. When that happens, it is an operational challenge to see the error from the log and then use a different disk or new peer (depending on the root cause of the error). Hence, I thought it is better to panic here to ease operational task. If you prefer to only log errors, I will update the code.

Reply:
What you mentioned above is the current behavior as well. No change in that as such, except that the error would now be logged.
If you want to change the behavior, IMO, the error state should be recorded such that the private data retrieval function should report the error to the caller. If you want to go that far, I would encourage to deal with that in a separate PR. But, as I mentioned before, `panic` is too brutal for this error.

Reply:
As this is a goroutine, the error propagation is quite hard to do. A larger refactoring is required as you mentioned. For now, I have reverted to the old behaviour + error logs.